### PR TITLE
Customer Trace Tester

### PR DESCRIPTION
**Pull request by mat-rumian**
The main goal of adding this tool is to give a customer simple way to check if everything is working fine end to end regarding traces.

Example Sumo UI view:
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,6 +76,26 @@ You can set Jaeger Go client env variables (such as `JAEGER_AGENT_HOST` or `JAEG
...
+  --env COLLECTOR_HOSTNAME=collection-sumologic-otelcol.sumologic \
+  --env TOTAL_TRACES=1 \
+  --env SPANS_PER_TRACE=10 \
+  -- customer-trace-tester
+```
+
+You can configure tool by setting env variables: such as `COLLECTOR_HOSTNAME` and stress-tester specific ones:
+- `COLLECTOR_HOSTNAME` (default=collection-sumologic-otelcol.sumologic) - the hostname/service of OpenTelemetry Collector
+- `TOTAL_TRACES` (default=1) - total number of traces to generate
+- `SPANS_PER_TRACE` (default=1) - number of spans per trace
```
Comment (pmalek-sumo) [inline README.md:97]:
```suggestion
You can configure this tool by setting the following env variables:

- `COLLECTOR_HOSTNAME` (default=collection-sumologic-otelcol.sumologic) - the hostname/service of OpenTelemetry Collector
- `TOTAL_TRACES` (default=1) - total number of traces to generate
- `SPANS_PER_TRACE` (default=1) - number of spans per trace
```

File: README.md [hunk 2]
```diff
@@ -76,6 +76,26 @@ You can set Jaeger Go client env variables (such as `JAEGER_AGENT_HOST` or `JAEG
 - `TOTAL_SPANS` (default=10000000) - total number of spans to generate
 - `SPANS_PER_MIN` (required) - rate of spans per minute (the tester will adjust the delay between iterations to reach such rate)
 
+### Customer Trace Tester
+There's a simple tool that generates a desired number of spans and traces and sends them using OpenTelemetry exporters.
```
Comment (pmalek-sumo) [inline README.md:80]:
```suggestion
### Customer Trace Tester

`stress-tester` is a simple tool that generates a desired number of spans and traces and sends them using OpenTelemetry exporters.
```

File: README.md [hunk 3]
```diff
@@ -76,6 +76,26 @@ You can set Jaeger Go client env variables (such as `JAEGER_AGENT_HOST` or `JAEG
...
+  --env TOTAL_TRACES=1 \
+  --env SPANS_PER_TRACE=10 \
+  -- customer-trace-tester
+```
+
+You can configure tool by setting env variables: such as `COLLECTOR_HOSTNAME` and stress-tester specific ones:
+- `COLLECTOR_HOSTNAME` (default=collection-sumologic-otelcol.sumologic) - the hostname/service of OpenTelemetry Collector
+- `TOTAL_TRACES` (default=1) - total number of traces to generate
+- `SPANS_PER_TRACE` (default=1) - number of spans per trace
+
```
Comment (pmalek-sumo) [inline README.md:98]: Can we add an exemplar output of this tool?

File: README.md [hunk 4]
```diff
@@ -76,6 +76,26 @@ You can set Jaeger Go client env variables (such as `JAEGER_AGENT_HOST` or `JAEG
...
+  --restart=Never -n sumologic \
+  --image sumologic/kubernetes-tools \
+  --serviceaccount='collection-sumologic' \
+  --env COLLECTOR_HOSTNAME=collection-sumologic-otelcol.sumologic \
+  --env TOTAL_TRACES=1 \
+  --env SPANS_PER_TRACE=10 \
+  -- customer-trace-tester
+```
+
+You can configure tool by setting env variables: such as `COLLECTOR_HOSTNAME` and stress-tester specific ones:
```
Comment (pmalek-sumo) [inline README.md:94]:
```suggestion

### Configuration

You can configure tool by setting env variables: such as `COLLECTOR_HOSTNAME` and stress-tester specific ones:
```

File: README.md (hunk 3 above)
Comment (mat-rumian) [reply README.md:102]: Good idea, added :)

File: README.md [hunk 5]
```diff
@@ -71,11 +71,69 @@ There's a simple tool that generates a desired number of spans per minute and se
...
+#### Configuration
+
 You can set Jaeger Go client env variables (such as `JAEGER_AGENT_HOST` or `JAEGER_COLLECTOR`) and stress-tester specific ones:
 
 - `TOTAL_SPANS` (default=10000000) - total number of spans to generate
 - `SPANS_PER_MIN` (required) - rate of spans per minute (the tester will adjust the delay between iterations to reach such rate)
 
+### Customer Trace Tester
+`customer-trace-tester` is simple tool that generates a desired number of spans and traces and sends them using OpenTelemetry exporters.
+Traces can be easily found with the `service=customer-trace-test-service` filter in the Sumo Logic web application.
```
Comment (pmalek-sumo) [inline README.md:83]:
```suggestion
### Customer Trace Tester

`customer-trace-tester` is simple tool that generates a desired number of spans and traces and sends them using OpenTelemetry exporters.
Traces can be easily found with the `service=customer-trace-test-service` filter in the Sumo Logic web application.
```

File: README.md [hunk 6]
```diff
@@ -71,11 +71,69 @@ There's a simple tool that generates a desired number of spans per minute and se
...
+  --restart=Never -n sumologic \
+  --image sumologic/kubernetes-tools \
+  --serviceaccount='collection-sumologic' \
+  --env COLLECTOR_HOSTNAME=collection-sumologic-otelcol.sumologic \
+  --env TOTAL_TRACES=1 \
+  --env SPANS_PER_TRACE=10 \
+  -- customer-trace-tester
+```
+#### Configuration
+
```
Comment (pmalek-sumo) [inline README.md:97]:
```suggestion

#### Configuration

```

File: README.md [hunk 7]
```diff
@@ -71,11 +71,69 @@ There's a simple tool that generates a desired number of spans per minute and se
...
+You can configure this tool by setting the following env variables:
+
+- `COLLECTOR_HOSTNAME` (default=collection-sumologic-otelcol.sumologic) - the hostname/service of OpenTelemetry Collector
+- `TOTAL_TRACES` (default=1) - total number of traces to generate
+- `SPANS_PER_TRACE` (default=10) - number of spans per trace
```
Comment (pmalek-sumo) [inline README.md:102]:
```suggestion
- `COLLECTOR_HOSTNAME` (default=`collection-sumologic-otelcol.sumologic`) - the hostname/service of OpenTelemetry Collector
- `TOTAL_TRACES` (default=`1`) - total number of traces to generate
- `SPANS_PER_TRACE` (default=`10`) - number of spans per trace
```

File: src/go/cmd/customer-trace-tester/main.go [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+var help bool
+
+func init() {
+	flag.BoolVar(&help, "help", false, "show help")
+}
+
+func main() {
+	flag.Parse()
+	if help {
+		fmt.Println("Customer trace-testing")
```
Comment (pmalek-sumo) [inline src/go/cmd/customer-trace-tester/main.go:234]: Can we add some more info here? I'd expect actual "help" when I run the binary with `-help` flag :)

Comment (mat-rumian) [reply src/go/cmd/customer-trace-tester/main.go]: Done :)

File: .markdownlint/style.rb [hunk 9]
```diff
@@ -18,7 +18,6 @@
 rule 'MD021'
 rule 'MD022'
 rule 'MD023'
-rule 'MD024'
```
Comment (pmalek-sumo) [inline .markdownlint/style.rb:21]: Don't remove it, we can use:  https://github.com/markdownlint/markdownlint/pull/200/files

```
rule 'MD024', :allow_different_nesting => true
```

File: README.md [hunk 10]
```diff
@@ -71,11 +71,73 @@ There's a simple tool that generates a desired number of spans per minute and se
...
+#### Configuration
+
 You can set Jaeger Go client env variables (such as `JAEGER_AGENT_HOST` or `JAEGER_COLLECTOR`) and stress-tester specific ones:
 
 - `TOTAL_SPANS` (default=10000000) - total number of spans to generate
 - `SPANS_PER_MIN` (required) - rate of spans per minute (the tester will adjust the delay between iterations to reach such rate)
 
+### Customer Trace Tester
+
+`customer-trace-tester` is simple tool that generates a desired number of spans and traces and sends them using OpenTelemetry exporters.
```
Comment (pmalek-sumo) [inline README.md:83]:
```suggestion
`customer-trace-tester` is a simple tool that generates a desired number of spans and traces and sends them using OpenTelemetry exporters.
```

**Review by pmalek-sumo — APPROVED** [review]
one nit and let's merge it 🚀

**Pull request merged**